### PR TITLE
add epics driver for MCC SSR08 digital out module

### PR DESCRIPTION
**Pull request by oleg-makarov**
Hi Mark,

here is an addition of epics driver for MCC SSR08 digital out module.

Oleg

### CODE REVIEW

File: configure/CONFIG_SITE [hunk 1]
```diff
@@ -20,6 +20,7 @@ CHECK_RELEASE = YES
 #   for a subset of the cross-compiled target architectures
 #   that Base is built for.
 #CROSS_COMPILER_TARGET_ARCHS = vxWorks-68040
+CROSS_COMPILER_TARGET_ARCHS =
```
Comment (MarkRivers) [inline configure/CONFIG_SITE:23]: There is no need for uncommenting that line, please revert to master.

File: configure/CONFIG_SITE.Common.linux-x86_64 [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Set this to YES if the hidapi package is installed for USB devices on Linux.
+HAVE_HIDAPI=YES
```
Comment (MarkRivers) [inline configure/CONFIG_SITE.Common.linux-x86_64:2]: There is need for this file, the HIDAPI library is no longer needed on Linux.  Please remove.

File: configure/RELEASE [hunk 3]
```diff
@@ -24,22 +24,26 @@ TEMPLATE_TOP=$(EPICS_BASE)/templates/makeBaseApp/top
 # If using the sequencer, point SNCSEQ at its top directory:
 #SNCSEQ=$(EPICS_BASE)/../modules/soft/seq
 
-SUPPORT=/corvette/home/epics/devel
+#SUPPORT=/corvette/home/epics/devel
```
Comment (MarkRivers) [inline configure/RELEASE:27]: These changes are specific to your site and should not be merged.  Please checkout master, make local edits, but do not commit and have as part of your PR.

File: measCompApp/Db/measCompBinaryOut.template [hunk 4]
```diff
@@ -1,6 +1,7 @@
 record(bo, "$(P)$(R)")
 {
-    field(PINI, "$(PINI=YES)")
+#   field(PINI, "$(PINI=YES)")
```
Comment (MarkRivers) [inline measCompApp/Db/measCompBinaryOut.template:3]: Adding DESC is not backwards compatible, you are forcing everyone to modify their files to include the DESC macro.

Comment (MarkRivers) [inline measCompApp/Db/measCompBinaryOut.template:3]: You also break backwards compatibility by commenting out the line with $(PINI=YES).  If you don't want PINI then just set PINI=NO in you file that loads this.

File: measCompApp/Db/measCompBinaryOut.template [hunk 5]
```diff
@@ -11,6 +12,7 @@ record(bo, "$(P)$(R)")
 
 record(bi, "$(P)$(R)_RBV")
 {
+    field(DESC, "$(DESC)")
```
Comment (MarkRivers) [inline measCompApp/Db/measCompBinaryOut.template:15]: Change this to $(DESC)= so it is optional.

File: measCompApp/Db/measCompLongOut.template [hunk 6]
```diff
@@ -3,15 +3,15 @@
 ###################################################################
 record(longout, "$(P)$(R)")
 {
-    field(PINI, "YES")
+#   field(PINI, "YES")
```
Comment (MarkRivers) [inline measCompApp/Db/measCompLongOut.template:6]: Change this to $(DESC)= so it is optional.

File: measCompApp/Db/measCompLongOut.template [hunk 7]
```diff
@@ -3,15 +3,15 @@
...
     field(DTYP, "asynUInt32Digital")
     field(OUT,  "@asynMask($(PORT),$(ADDR),$(MASK))DIGITAL_OUTPUT")
 }
 
 record(longin, "$(P)$(R)_RBV")
 {
     field(DTYP, "asynUInt32Digital")
-    field(INP,  "@asynMask($(PORT),$(ADDR),$(MASK))DIGITAL_OUTPUT")
+    field(INP,  "@asynMask($(PORT),$(ADDR),$(MASK))DIGITAL_INPUT")
```
Comment (MarkRivers) [inline measCompApp/Db/measCompLongOut.template:14]: Changing DIGITAL_OUTPUT to DIGITAL_INPUT is incorrect.  DIGITAL_OUTPUT is a readback of the output register.  DIGITAL_INPUT is reading the input register, which is different. Did you observe and issue that motivated this change?

File: measCompApp/src/Makefile [hunk 8]
```diff
@@ -60,6 +60,7 @@ measCompApp_LIBS += sscan
 measCompApp_LIBS += autosave
 measCompApp_LIBS += asyn
 measCompApp_LIBS += seq pv
+measCompApp_LIBS += qsrv pvAccessIOC pvAccessCA pvAccess pvData
```
Comment (MarkRivers) [inline measCompApp/src/Makefile:63]: There is no need to build this IOC with PV access.  If you want to build you real IOC with them that is fine.  Please revert to master.

File: measCompApp/src/drvMultiFunction.cpp [hunk 9]
```diff
@@ -817,7 +833,7 @@ MultiFunction::MultiFunction(const char *portName, const char *uniqueID, int max
       ASYN_MULTIDEVICE | ASYN_CANBLOCK, 1, /* ASYN_CANBLOCK=1, ASYN_MULTIDEVICE=1, autoConnect=1 */
       0, 0),  /* Default priority and stack size */
     pollTime_(DEFAULT_POLL_TIME),
-    forceCallback_(1),
+    forceCallback_{1,1,1},
```
Comment (MarkRivers) [inline measCompApp/src/drvMultiFunction.cpp:836]: This initialization should probably be done in the constructor using a loop over MAX_IO_PORTS so it changes automatically when that number is increased.

File: measCompApp/src/measCompAppInclude.dbd [hunk 10]
```diff
@@ -1,4 +1,6 @@
 include "base.dbd"
+include "PVAServerRegister.dbd"
+include "qsrv.dbd"
```
Comment (MarkRivers) [inline measCompApp/src/measCompAppInclude.dbd:3]: Remove these lines, they are not needed for test IOC and won't work on older versions of base.

File: configure/CONFIG_SITE.Common.linux-x86_64 (hunk 2 above)
Comment (oleg-makarov) [reply configure/CONFIG_SITE.Common.linux-x86_64]: file removed

File: configure/RELEASE (hunk 3 above)
Comment (oleg-makarov) [reply configure/RELEASE:27]: OK

File: measCompApp/Db/measCompBinaryOut.template (hunk 4 above)
Comment (oleg-makarov) [reply measCompApp/Db/measCompBinaryOut.template]: OK, now DESC has an empty default value

Comment (oleg-makarov) [reply measCompApp/Db/measCompBinaryOut.template]: ok

File: measCompApp/Db/measCompLongOut.template (hunk 6 above)
Comment (oleg-makarov) [reply measCompApp/Db/measCompLongOut.template:6]: ok

File: measCompApp/Db/measCompLongOut.template (hunk 7 above)
Comment (oleg-makarov) [reply measCompApp/Db/measCompLongOut.template]: reverted back

File: measCompApp/src/Makefile (hunk 8 above)
Comment (oleg-makarov) [reply measCompApp/src/Makefile]: ok, commented out this line

File: measCompApp/src/drvMultiFunction.cpp (hunk 9 above)
Comment (oleg-makarov) [reply measCompApp/src/drvMultiFunction.cpp]: ok, moved this initialization into the constructor

**Comment by oleg-makarov** [comment]
Thank you, Mark!

**Pull request closed without merging**